### PR TITLE
[5.0] Use the current user function in view instead of from the app

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Uses the current user from the view in classes and template files.

### Testing Instructions
Open any view from a touched file and check if there is no error.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/modules/mod_submenu/tmpl/default.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Router\Route;
 
-$user = \Joomla\CMS\Factory::getApplication()->getIdentity();
+$user = \Joomla\CMS\$this->getCurrentUser();
```
Comment (HLeithner) [inline administrator/modules/mod_submenu/tmpl/default.php]:
```suggestion
$user = $this->getCurrentUser();
```

File: administrator/components/com_users/src/Model/MethodModel.php [hunk 2]
```diff
@@ -91,7 +91,7 @@ public function methodExists(string $method): bool
     public function getRenderOptions(?User $user = null): SetupRenderOptions
     {
         if (is_null($user)) {
-            $user = Factory::getApplication()->getIdentity() ?: $this->getCurrentUser();
+            $user = $this->getCurrentUser() ?: Factory::getApplication()->getIdentity();
```
Comment (HLeithner) [inline administrator/components/com_users/src/Model/MethodModel.php]:
```suggestion
            $user = $this->getCurrentUser();
```
getCurrentUser automatically falls back to the application

**Review by HLeithner — CHANGES_REQUESTED** [review]

**Comment by HLeithner** [comment]
Thanks

**Comment by brianteeman** [comment]
and this is why we shouldnt merge untested code see #41414

**Pull request merged**